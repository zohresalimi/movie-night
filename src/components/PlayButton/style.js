import styled from "styled-components";

export const PlayBtn = styled.a`
  position: absolute;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 60px;
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 50%;
  top: 50%;
  left: 50%;
  margin: -30px 0 0 -30px;
  z-index: 3;
  font-size: 30px;
  color: #fd6060;
  transition: 0.4s ease;
  &:before {
    content: "";
    position: absolute;
    display: block;
    width: 48px;
    height: 48px;
    top: 50%;
    left: 50%;
    margin: -24px 0 0 -24px;
    border-radius: 50%;
    background-color: #fff;
    z-index: 1;
  }
  & i {
    position: relative;
    z-index: 2;
    margin: 2px 0 0 3px;
  }
  &:hover {
    background-color: rgba(253, 96, 96, 0.6);
    color: #fd6060;
  }
`;
