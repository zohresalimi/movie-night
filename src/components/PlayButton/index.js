import React from "react";
import { Icon } from "semantic-ui-react";

import { PlayBtn } from "./style";

function PlayButton({ setPlayVideo }) {
  return (
    <div onClick={() => setPlayVideo(true)} data-testid="play-btn">
      <PlayBtn>
        <Icon name="play" size="mini" />
      </PlayBtn>
    </div>
  );
}

export default PlayButton;
