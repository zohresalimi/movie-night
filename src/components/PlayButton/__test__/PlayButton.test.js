import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PlayButton from "../";

describe("PlayButton Component Testing", () => {
  test("take snapshot", async () => {
    const { container, getByTestId } = await render(<PlayButton />);
    expect(getByTestId("play-btn")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("check setPlayVideo should be called", async () => {
    const mockSetPlayVideo = jest.fn(() => true);
    const { getByTestId } = await render(
      <PlayButton setPlayVideo={mockSetPlayVideo} />
    );
    fireEvent.click(getByTestId("play-btn"));
    expect(mockSetPlayVideo).toHaveBeenCalled();
  });
});
