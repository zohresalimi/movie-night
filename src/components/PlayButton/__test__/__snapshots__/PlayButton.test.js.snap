// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton Component Testing take snapshot 1`] = `
<button
  data-testid="play-btn"
>
  <a
    class="sc-bdfBwQ gaGgLj"
  >
    <i
      aria-hidden="true"
      class="play mini icon"
    />
  </a>
</button>
`;
